net/net.sh

```diff
@@ -232,8 +232,10 @@ build() {
   echo "Build took $SECONDS seconds"
 }
 
-SOLANA_HOME="\$HOME/solana"
-CARGO_BIN="\$HOME/.cargo/bin"
+# shellcheck disable=SC2088
+SOLANA_HOME="~/solana"
+# shellcheck disable=SC2088
+CARGO_BIN="~/.cargo/bin"
```

Comment:
```suggestion
remoteHome="$(ssh "${sshOptions[@]}" "$ipAddress" "echo \$HOME")"
SOLANA_HOME="${remoteHome}/solana"
CARGO_BIN="${remoteHome}/.cargo/bin"
```

wdyt of this?  seems like the solution with the least dependence on knowledge of bash magic

Reply:
only as an isolated ssh command, not a full net.sh run

Reply:
It _might_ have issues with `rsync` command. Trying it right now.

Reply:
Getting the following error with this change

```
./net.sh start
ssh: Could not resolve hostname : Name or service not known
```

Reply:
@t-nelson the `remoteHome` command had to be moved into individual functions, as the `ipAddress` is available only there. The latest commit has the changes.

Reply:
@t-nelson the latest commit reduces the replicode. The local variable definitions have to stay, but the call to `ssh ...` has been moved to a common function.

Reply:
> @t-nelson the latest commit reduces the replicode. The local variable definitions have to stay, but the call to `ssh ...` has been moved to a common function.

thanks! prolly about as good as we can do without a more invasive refactor